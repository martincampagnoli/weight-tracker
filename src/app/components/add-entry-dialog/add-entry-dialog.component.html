<h2 mat-dialog-title>Add entry</h2>
<mat-dialog-content>
  <app-add-entry [dialogRef]="dialogRef" />
</mat-dialog-content>
