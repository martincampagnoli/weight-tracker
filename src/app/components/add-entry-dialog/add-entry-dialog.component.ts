import { ChangeDetectionStrategy, Component, inject } from '@angular/core';
import { MatButtonModule } from '@angular/material/button';
import {
  MatDialogTitle,
  MatDialogContent,
  MatDialogRef,
} from '@angular/material/dialog';
import { AddEntryComponent } from '../add-entry/add-entry.component';

@Component({
  selector: 'add-entry-dialog',
  templateUrl: 'add-entry-dialog.component.html',
  standalone: true,
  imports: [
    MatButtonModule,
    MatDialogTitle,
    MatDialogContent,
    AddEntryComponent,
  ],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class AddEntryDialogComponent {
  readonly dialogRef = inject(MatDialogRef<AddEntryDialogComponent>);
}
