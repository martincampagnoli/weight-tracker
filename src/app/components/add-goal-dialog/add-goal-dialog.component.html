<h2 mat-dialog-title>Add goal</h2>
<mat-dialog-content>
  <app-add-goal [dialogRef]="dialogRef" />
</mat-dialog-content>
