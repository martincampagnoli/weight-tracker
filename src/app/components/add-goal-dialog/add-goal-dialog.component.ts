import { ChangeDetectionStrategy, Component, inject } from '@angular/core';
import { MatButtonModule } from '@angular/material/button';
import {
  MatDialogTitle,
  MatDialogContent,
  MatDialogRef,
} from '@angular/material/dialog';
import { AddGoalComponent } from '../add-goal/add-goal.component';

@Component({
  selector: 'add-goal-dialog',
  templateUrl: 'add-goal-dialog.component.html',
  standalone: true,
  imports: [
    MatButtonModule,
    MatDialogTitle,
    MatDialogContent,
    AddGoalComponent,
  ],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class AddGoalDialogComponent {
  readonly dialogRef = inject(MatDialogRef<AddGoalDialogComponent>);
}
