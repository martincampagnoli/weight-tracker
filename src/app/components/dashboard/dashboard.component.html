<div class="container dashboard-container">
  <section class="mx-auto mb-2 dashboard-section">
    <div class="row align-items-center flex-wrap">
      <div class="col-md-6 p-2 d-flex align-items-center gap-2">
        <span class="entries-count text-nowrap min-width-90">
          {{ filteredData().length + " entries" }}
        </span>
        @if (goal()) {
        <app-progress-bar />
        } @else {
        <div class="w-100 d-flex justify-content-center">
          <span class="goal-message"
            >Set a weight goal to improve your health...</span
          >
        </div>
        }
      </div>
      <div class="col-md-6 p-2">
        <div class="row justify-content-md-end justify-content-center">
          <div class="col-12 col-md-4 p-2">
            <button
              (click)="openDialog('viewProgress')"
              color="primary"
              mat-button
              mat-raised-button
              class="w-100"
            >
              <mat-icon>show_chart</mat-icon>
              <span class="d-none d-md-inline"> View progress</span>
              <span class="d-md-none"> Progress</span>
            </button>
          </div>
          <div class="col-12 col-md-4 p-2">
            <button
              (click)="openDialog('setWeightGoal')"
              color="primary"
              mat-button
              mat-raised-button
              class="w-100"
            >
              <mat-icon>sports_score</mat-icon>
              <span class="d-none d-md-inline"> Set weight goal</span>
              <span class="d-md-none"> Goal</span>
            </button>
          </div>
          <div class="col-12 col-md-4 p-2">
            <button
              (click)="openDialog('addEntry')"
              color="primary"
              mat-button
              mat-raised-button
              class="w-100"
            >
              <mat-icon>add</mat-icon>
              <span class="d-none d-md-inline"> Add new entry</span>
              <span class="d-md-none"> Add</span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div class="row">
        @for ( item of filteredData() | slice: lowValue : highValue; track
        item.id ) {
        <div class="col-md-4 p-2 entry-grid-item">
          <app-entry [entry]="item"></app-entry>
        </div>
        }
      </div>
    </div>
    <div class="row mb-3">
      @if (filteredData().length > 3) {
      <mat-paginator
        [length]="filteredData().length"
        [pageSize]="pageSize"
        (page)="getPaginatorData($event)"
        [pageSizeOptions]="pageSizeOptions"
        class="col-12 bg-transparent"
      >
      </mat-paginator>
      }
    </div>
  </section>
</div>
