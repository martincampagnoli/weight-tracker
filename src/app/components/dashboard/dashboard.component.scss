.min-width-90 {
  min-width: 90px;
}

.goal-message {
  font-style: italic;
}
.dashboard-container {
  margin-top: 32px;
  border: solid 1px;
  border-radius: 4px;
  border-color: rgb(217, 224, 232);
}

.dashboard-section {
  padding-right: 16px;
  padding-left: 16px;
  padding-top: 5px;
}

.entries-count {
  font-size: x-large;
  font-weight: 400;
}

.entry-grid-item {
  overflow: hidden;
}
