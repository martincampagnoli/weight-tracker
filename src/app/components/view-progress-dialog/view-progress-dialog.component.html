<h2 mat-dialog-title>View Progress</h2>
<mat-dialog-content>
  <app-view-progress [dialogRef]="dialogRef" />
</mat-dialog-content>
