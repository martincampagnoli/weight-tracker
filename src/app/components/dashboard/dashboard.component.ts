import { CommonModule } from '@angular/common';
import {
  ChangeDetectionStrategy,
  Component,
  inject,
  Type,
} from '@angular/core';
import { takeUntilDestroyed, toSignal } from '@angular/core/rxjs-interop';
import { MatIcon } from '@angular/material/icon';
import { MatButton } from '@angular/material/button';
import { MatPaginator, PageEvent } from '@angular/material/paginator';
import { Store } from '@ngrx/store';
import { getData } from 'src/app/store/default';
import { EntryComponent } from '../entry/entry.component';
import { Entry } from 'src/app/models/Entry';
import * as reducers from 'src/app/store/default';
import { MatDialog } from '@angular/material/dialog';
import { AddEntryDialogComponent } from '../add-entry-dialog/add-entry-dialog.component';
import { ViewProgressDialogComponent } from '../view-progress-dialog/view-progress-dialog.component';

@Component({
  selector: 'app-dashboard',
  templateUrl: './dashboard.component.html',
  styleUrls: ['./dashboard.component.scss'],
  standalone: true,
  changeDetection: ChangeDetectionStrategy.OnPush,
  imports: [CommonModule, MatIcon, MatButton, MatPaginator, EntryComponent],
})
export class DashboardComponent {
  readonly dialog = inject(MatDialog);
  private store: Store = inject(Store);
  protected filteredData = toSignal(this.store.select(reducers.getDataState), {
    initialValue: [],
  });

  lowValue = 0;
  highValue = 3;
  pageSize = 3;
  pageSizeOptions = [3, 9, 30, 90];

  constructor() {
    this.store.dispatch(getData());
  }

  getPaginatorData(event: PageEvent): PageEvent {
    this.lowValue = event.pageIndex * event.pageSize;
    this.highValue = this.lowValue + event.pageSize;
    return event;
  }

  openDialog(type: string): void {
    let component: Type<AddEntryDialogComponent | ViewProgressDialogComponent>;
    switch (type) {
      case 'addEntry':
        component = AddEntryDialogComponent;
        break;
      case 'viewProgress':
        component = ViewProgressDialogComponent;
        break;
      default:
        component = AddEntryDialogComponent;
    }

    this.dialog.open(component, {
      width: '400px',
      enterAnimationDuration: '0ms',
      exitAnimationDuration: '0ms',
    });
  }
}
