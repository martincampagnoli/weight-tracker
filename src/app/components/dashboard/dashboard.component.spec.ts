import { ComponentFixture, TestBed } from '@angular/core/testing';
import { MatDialog } from '@angular/material/dialog';
import { MockStore, provideMockStore } from '@ngrx/store/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { PageEvent } from '@angular/material/paginator';
import { of } from 'rxjs';
import { DashboardComponent } from './dashboard.component';
import { Entry } from 'src/app/models/Entry';
import * as reducers from 'src/app/store/default';
import { AddEntryDialogComponent } from '../add-entry-dialog/add-entry-dialog.component';
import { ViewProgressDialogComponent } from '../view-progress-dialog/view-progress-dialog.component';
import { GoalService } from 'src/app/services/goal.service';
import { AddGoalDialogComponent } from '../add-goal-dialog/add-goal-dialog.component';

describe('DashboardComponent', () => {
  let component: DashboardComponent;
  let fixture: ComponentFixture<DashboardComponent>;
  let store: MockStore;
  let mockDialog: jest.Mocked<MatDialog>;
  const mockGoalService = {
    calculateGoalProgress: jest.fn().mockReturnValue(75),
  };

  const mockEntries: Entry[] = [
    { id: 1, weight: 70, date: '01-01-2024', description: 'Start' },
    { id: 2, weight: 68, date: '15-01-2024', description: 'Progress' },
    { id: 3, weight: 65, date: '01-02-2024', description: 'Goal reached' },
    { id: 4, weight: 67, date: '15-02-2024', description: 'Maintenance' },
    { id: 5, weight: 66, date: '01-03-2024', description: 'Continued' },
  ];

  const initialState = {} as any;

  beforeEach(async () => {
    mockDialog = {
      open: jest.fn().mockReturnValue({
        afterClosed: () => of(true),
      }),
    } as any;

    await TestBed.configureTestingModule({
      imports: [DashboardComponent, NoopAnimationsModule],
      providers: [
        { provide: GoalService, useValue: mockGoalService },
        provideMockStore({ initialState }),
        {
          provide: MatDialog,
          useValue: mockDialog,
        },
      ],
    }).compileComponents();

    store = TestBed.inject(MockStore);
    store.overrideSelector(reducers.getDataState, mockEntries);
    store.overrideSelector(reducers.getTargetWeight, 75);

    fixture = TestBed.createComponent(DashboardComponent);
    component = fixture.componentInstance;
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  describe('Component Creation', () => {
    it('should create', () => {
      expect(component).toBeTruthy();
    });

    it('should match snapshot', () => {
      fixture.detectChanges();
      expect(fixture).toMatchSnapshot();
    });
  });

  describe('Initialization', () => {
    it('should initialize with correct default values', () => {
      expect(component.lowValue).toBe(0);
      expect(component.highValue).toBe(3);
      expect(component.pageSize).toBe(3);
      expect(component.pageSizeOptions).toEqual([3, 9, 30, 90]);
    });

    it('should dispatch getData action on construction', () => {
      const dispatchSpy = jest.spyOn(store, 'dispatch');

      const newFixture = TestBed.createComponent(DashboardComponent);

      expect(dispatchSpy).toHaveBeenCalledWith(reducers.getData());
    });

    it('should subscribe to data state', () => {
      fixture.detectChanges();

      expect(component['filteredData']()).toEqual(mockEntries);
    });
  });

  describe('Pagination', () => {
    beforeEach(() => {
      fixture.detectChanges();
    });

    it('should handle page events correctly', () => {
      const pageEvent: PageEvent = {
        pageIndex: 1,
        pageSize: 3,
        length: 5,
      };

      const result = component.getPaginatorData(pageEvent);

      expect(component.lowValue).toBe(3);
      expect(component.highValue).toBe(6);
      expect(result).toEqual(pageEvent);
    });

    it('should calculate pagination correctly for different page sizes', () => {
      const pageEvent: PageEvent = {
        pageIndex: 0,
        pageSize: 9,
        length: 5,
      };

      component.getPaginatorData(pageEvent);

      expect(component.lowValue).toBe(0);
      expect(component.highValue).toBe(9);
    });
  });

  describe('Dialog Opening', () => {
    beforeEach(() => {
      fixture.detectChanges();
    });

    it('should open AddEntryDialogComponent for addEntry type', () => {
      component.openDialog('addEntry');

      expect(mockDialog.open).toHaveBeenCalledWith(AddEntryDialogComponent, {
        width: '400px',
        enterAnimationDuration: '0ms',
        exitAnimationDuration: '0ms',
      });
    });

    it('should open ViewProgressDialogComponent for viewProgress type', () => {
      component.openDialog('viewProgress');

      expect(mockDialog.open).toHaveBeenCalledWith(
        ViewProgressDialogComponent,
        {
          width: '400px',
          enterAnimationDuration: '0ms',
          exitAnimationDuration: '0ms',
        }
      );
    });

    it('should open AddEntryDialogComponent for unknown type (default)', () => {
      component.openDialog('unknown');

      expect(mockDialog.open).toHaveBeenCalledWith(AddEntryDialogComponent, {
        width: '400px',
        enterAnimationDuration: '0ms',
        exitAnimationDuration: '0ms',
      });
    });

    it('should open AddGoalDialogComponent for setWeightGoal type', () => {
      component.openDialog('setWeightGoal');

      expect(mockDialog.open).toHaveBeenCalledWith(AddGoalDialogComponent, {
        width: '400px',
        enterAnimationDuration: '0ms',
        exitAnimationDuration: '0ms',
      });
    });
  });

  describe('Template Rendering', () => {
    beforeEach(() => {
      store.overrideSelector(reducers.getDataState, mockEntries);
      store.overrideSelector(reducers.getTargetWeight, 65);
      store.refreshState();
      fixture.detectChanges();
    });

    it('should render entries count', () => {
      const entriesCount =
        fixture.debugElement.nativeElement.querySelector('.entries-count');
      expect(entriesCount.textContent.trim()).toBe('5 entries');
    });

    it('should render action buttons', () => {
      const buttons =
        fixture.debugElement.nativeElement.querySelectorAll('button');
      expect(buttons.length).toBeGreaterThanOrEqual(3);
    });

    it('should render paginator when entries > 3', () => {
      const paginator =
        fixture.debugElement.nativeElement.querySelector('mat-paginator');
      expect(paginator).toBeTruthy();
    });
  });

  describe('Data Updates', () => {
    it('should update filteredData when store data changes', () => {
      const newEntries: Entry[] = [
        { id: 6, weight: 64, date: '15-03-2024', description: 'New entry' },
      ];

      store.overrideSelector(reducers.getDataState, newEntries);
      store.refreshState();
      fixture.detectChanges();

      expect(component['filteredData']()).toEqual(newEntries);
    });
  });
});
