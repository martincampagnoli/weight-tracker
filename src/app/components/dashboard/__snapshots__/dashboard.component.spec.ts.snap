// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardComponent Component Creation should match snapshot 1`] = `
<app-dashboard
  dialog={[Function Object]}
  filteredData={[Function Function]}
  highValue={[Function Number]}
  lowValue="0"
  pageSize={[Function Number]}
  pageSizeOptions={[Function Array]}
  store={[Function _MockStore]}
>
  <div
    class="container dashboard-container"
  >
    <section
      class="mx-auto mb-2 dashboard-section"
    >
      <div
        class="row align-items-center flex-wrap"
      >
        <div
          class="col-md-6 p-2"
        >
          <span
            class="entries-count"
          >
             5 entries 
          </span>
        </div>
        <div
          class="col-md-6 p-2"
        >
          <div
            class="row justify-content-md-end justify-content-center"
          >
            <div
              class="col-12 col-md-4 p-2"
            >
              <button
                class="mdc-button mat-mdc-button-base w-100 mdc-button--raised mat-mdc-raised-button mat-primary _mat-animation-noopable"
                color="primary"
                mat-button=""
                mat-raised-button=""
                mat-ripple-loader-class-name="mat-mdc-button-ripple"
                mat-ripple-loader-uninitialized=""
              >
                <span
                  class="mat-mdc-button-persistent-ripple mdc-button__ripple"
                />
                <mat-icon
                  aria-hidden="true"
                  class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color"
                  data-mat-icon-type="font"
                  role="img"
                >
                  show_chart
                </mat-icon>
                <span
                  class="mdc-button__label"
                >
                  <span
                    class="d-none d-md-inline"
                  >
                     View progress
                  </span>
                  <span
                    class="d-md-none"
                  >
                     Progress
                  </span>
                </span>
                <span
                  class="mat-focus-indicator"
                />
                <span
                  class="mat-mdc-button-touch-target"
                />
              </button>
            </div>
            <div
              class="col-12 col-md-4 p-2"
            >
              <button
                class="mdc-button mat-mdc-button-base w-100 mdc-button--raised mat-mdc-raised-button mat-primary _mat-animation-noopable"
                color="primary"
                mat-button=""
                mat-raised-button=""
                mat-ripple-loader-class-name="mat-mdc-button-ripple"
                mat-ripple-loader-uninitialized=""
              >
                <span
                  class="mat-mdc-button-persistent-ripple mdc-button__ripple"
                />
                <mat-icon
                  aria-hidden="true"
                  class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color"
                  data-mat-icon-type="font"
                  role="img"
                >
                  sports_score
                </mat-icon>
                <span
                  class="mdc-button__label"
                >
                  <span
                    class="d-none d-md-inline"
                  >
                     Set weight goal
                  </span>
                  <span
                    class="d-md-none"
                  >
                     Goal
                  </span>
                </span>
                <span
                  class="mat-focus-indicator"
                />
                <span
                  class="mat-mdc-button-touch-target"
                />
              </button>
            </div>
            <div
              class="col-12 col-md-4 p-2"
            >
              <button
                class="mdc-button mat-mdc-button-base w-100 mdc-button--raised mat-mdc-raised-button mat-primary _mat-animation-noopable"
                color="primary"
                mat-button=""
                mat-raised-button=""
                mat-ripple-loader-class-name="mat-mdc-button-ripple"
                mat-ripple-loader-uninitialized=""
              >
                <span
                  class="mat-mdc-button-persistent-ripple mdc-button__ripple"
                />
                <mat-icon
                  aria-hidden="true"
                  class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color"
                  data-mat-icon-type="font"
                  role="img"
                >
                  add
                </mat-icon>
                <span
                  class="mdc-button__label"
                >
                  <span
                    class="d-none d-md-inline"
                  >
                     Add new entry
                  </span>
                  <span
                    class="d-md-none"
                  >
                     Add
                  </span>
                </span>
                <span
                  class="mat-focus-indicator"
                />
                <span
                  class="mat-mdc-button-touch-target"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="row"
        >
          <div
            class="col-md-4 p-2 entry-grid-item"
          >
            <app-entry>
              <mat-card
                appearance="outlined"
                class="mat-mdc-card mdc-card entry-card mat-mdc-card-outlined mdc-card--outlined"
              >
                <mat-card-header
                  class="mat-mdc-card-header entry-header"
                >
                  <div
                    class="mat-mdc-card-header-text"
                  >
                    <mat-card-title
                      class="mat-mdc-card-title entry-title"
                    >
                      <span>
                        70 kgs
                      </span>
                    </mat-card-title>
                    <mat-card-subtitle
                      class="mat-mdc-card-subtitle entry-subtitle"
                    >
                      <span>
                        01-01-2024
                      </span>
                    </mat-card-subtitle>
                  </div>
                  <div
                    class="header-icons"
                  >
                    <span
                      aria-expanded="false"
                      aria-haspopup="menu"
                      class="mat-mdc-menu-trigger"
                    >
                      <mat-icon
                        aria-hidden="true"
                        class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color"
                        data-mat-icon-type="font"
                        role="img"
                      >
                        more_vert
                      </mat-icon>
                    </span>
                  </div>
                </mat-card-header>
                <mat-card-content
                  class="mat-mdc-card-content entry-content px-3"
                >
                  <p
                    class="entry-description mb-4"
                  >
                     Start 
                  </p>
                  <hr />
                </mat-card-content>
                <mat-card-actions
                  class="mat-mdc-card-actions mdc-card__actions d-flex justify-content-end ps-2 pe-3 pt-1 entry-actions"
                >
                  <span
                    class="push-right"
                  >
                    ID: 1
                  </span>
                </mat-card-actions>
              </mat-card>
              <mat-menu>
              </mat-menu>
            </app-entry>
          </div>
          <div
            class="col-md-4 p-2 entry-grid-item"
          >
            <app-entry>
              <mat-card
                appearance="outlined"
                class="mat-mdc-card mdc-card entry-card mat-mdc-card-outlined mdc-card--outlined"
              >
                <mat-card-header
                  class="mat-mdc-card-header entry-header"
                >
                  <div
                    class="mat-mdc-card-header-text"
                  >
                    <mat-card-title
                      class="mat-mdc-card-title entry-title"
                    >
                      <span>
                        68 kgs
                      </span>
                    </mat-card-title>
                    <mat-card-subtitle
                      class="mat-mdc-card-subtitle entry-subtitle"
                    >
                      <span>
                        15-01-2024
                      </span>
                    </mat-card-subtitle>
                  </div>
                  <div
                    class="header-icons"
                  >
                    <span
                      aria-expanded="false"
                      aria-haspopup="menu"
                      class="mat-mdc-menu-trigger"
                    >
                      <mat-icon
                        aria-hidden="true"
                        class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color"
                        data-mat-icon-type="font"
                        role="img"
                      >
                        more_vert
                      </mat-icon>
                    </span>
                  </div>
                </mat-card-header>
                <mat-card-content
                  class="mat-mdc-card-content entry-content px-3"
                >
                  <p
                    class="entry-description mb-4"
                  >
                     Progress 
                  </p>
                  <hr />
                </mat-card-content>
                <mat-card-actions
                  class="mat-mdc-card-actions mdc-card__actions d-flex justify-content-end ps-2 pe-3 pt-1 entry-actions"
                >
                  <span
                    class="push-right"
                  >
                    ID: 2
                  </span>
                </mat-card-actions>
              </mat-card>
              <mat-menu>
              </mat-menu>
            </app-entry>
          </div>
          <div
            class="col-md-4 p-2 entry-grid-item"
          >
            <app-entry>
              <mat-card
                appearance="outlined"
                class="mat-mdc-card mdc-card entry-card mat-mdc-card-outlined mdc-card--outlined"
              >
                <mat-card-header
                  class="mat-mdc-card-header entry-header"
                >
                  <div
                    class="mat-mdc-card-header-text"
                  >
                    <mat-card-title
                      class="mat-mdc-card-title entry-title"
                    >
                      <span>
                        65 kgs
                      </span>
                    </mat-card-title>
                    <mat-card-subtitle
                      class="mat-mdc-card-subtitle entry-subtitle"
                    >
                      <span>
                        01-02-2024
                      </span>
                    </mat-card-subtitle>
                  </div>
                  <div
                    class="header-icons"
                  >
                    <span
                      aria-expanded="false"
                      aria-haspopup="menu"
                      class="mat-mdc-menu-trigger"
                    >
                      <mat-icon
                        aria-hidden="true"
                        class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color"
                        data-mat-icon-type="font"
                        role="img"
                      >
                        more_vert
                      </mat-icon>
                    </span>
                  </div>
                </mat-card-header>
                <mat-card-content
                  class="mat-mdc-card-content entry-content px-3"
                >
                  <p
                    class="entry-description mb-4"
                  >
                     Goal reached 
                  </p>
                  <hr />
                </mat-card-content>
                <mat-card-actions
                  class="mat-mdc-card-actions mdc-card__actions d-flex justify-content-end ps-2 pe-3 pt-1 entry-actions"
                >
                  <span
                    class="push-right"
                  >
                    ID: 3
                  </span>
                </mat-card-actions>
              </mat-card>
              <mat-menu>
              </mat-menu>
            </app-entry>
          </div>
        </div>
      </div>
      <div
        class="row mb-3"
      >
        <mat-paginator
          class="mat-mdc-paginator col-12 bg-transparent"
          role="group"
        >
          <div
            class="mat-mdc-paginator-outer-container"
          >
            <div
              class="mat-mdc-paginator-container"
            >
              <div
                class="mat-mdc-paginator-page-size"
              >
                <div
                  class="mat-mdc-paginator-page-size-label"
                  id="mat-paginator-page-size-label-a0"
                >
                   Items per page: 
                </div>
                <mat-form-field
                  class="mat-mdc-form-field mat-mdc-paginator-page-size-select mat-mdc-form-field-type-mat-select mat-form-field-appearance-outline mat-primary"
                >
                  <div
                    class="mat-mdc-text-field-wrapper mdc-text-field mdc-text-field--outlined mdc-text-field--no-label"
                  >
                    <div
                      class="mat-mdc-form-field-flex"
                    >
                      <div
                        class="mdc-notched-outline mdc-notched-outline--no-label"
                        matformfieldnotchedoutline=""
                      >
                        <div
                          class="mat-mdc-notch-piece mdc-notched-outline__leading"
                        />
                        <div
                          class="mat-mdc-notch-piece mdc-notched-outline__notch"
                        >
                        </div>
                        <div
                          class="mat-mdc-notch-piece mdc-notched-outline__trailing"
                        />
                      </div>
                      <div
                        class="mat-mdc-form-field-infix"
                      >
                        <mat-select
                          aria-disabled="false"
                          aria-expanded="false"
                          aria-haspopup="listbox"
                          aria-invalid="false"
                          aria-labelledby=" mat-paginator-page-size-label-a0"
                          aria-required="false"
                          class="mat-mdc-select mat-mdc-select-empty"
                          hidesingleselectionindicator=""
                          id="mat-select-a0"
                          role="combobox"
                          tabindex="0"
                        >
                          <div
                            cdk-overlay-origin=""
                            class="mat-mdc-select-trigger"
                          >
                            <div
                              class="mat-mdc-select-value"
                              id="mat-select-value-a0"
                            >
                              <span
                                class="mat-mdc-select-placeholder mat-mdc-select-min-line"
                              >
                              </span>
                            </div>
                            <div
                              class="mat-mdc-select-arrow-wrapper"
                            >
                              <div
                                class="mat-mdc-select-arrow"
                              >
                                <svg
                                  aria-hidden="true"
                                  focusable="false"
                                  height="24px"
                                  viewBox="0 0 24 24"
                                  width="24px"
                                >
                                  <path
                                    d="M7 10l5 5 5-5z"
                                  />
                                </svg>
                              </div>
                            </div>
                          </div>
                        </mat-select>
                        <div
                          class="mat-mdc-paginator-touch-target"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    aria-atomic="true"
                    aria-live="polite"
                    class="mat-mdc-form-field-subscript-wrapper mat-mdc-form-field-bottom-align"
                  >
                    <div
                      class="mat-mdc-form-field-hint-wrapper"
                    >
                      <div
                        class="mat-mdc-form-field-hint-spacer"
                      />
                    </div>
                  </div>
                </mat-form-field>
              </div>
              <div
                class="mat-mdc-paginator-range-actions"
              >
                <div
                  aria-live="polite"
                  class="mat-mdc-paginator-range-label"
                >
                   1 – 3 of 5 
                </div>
                <button
                  aria-disabled="true"
                  aria-label="Previous page"
                  class="mdc-icon-button mat-mdc-icon-button mat-mdc-button-base mat-mdc-tooltip-trigger mat-mdc-paginator-navigation-previous mat-mdc-button-disabled mat-mdc-button-disabled-interactive mat-unthemed _mat-animation-noopable mat-mdc-tooltip-disabled"
                  disabledinteractive=""
                  mat-ripple-loader-centered=""
                  mat-ripple-loader-class-name="mat-mdc-button-ripple"
                  mat-ripple-loader-disabled=""
                  mat-ripple-loader-uninitialized=""
                  maticonbutton=""
                  mattooltipposition="above"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="mat-mdc-button-persistent-ripple mdc-icon-button__ripple"
                  />
                  <svg
                    aria-hidden="true"
                    class="mat-mdc-paginator-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                    />
                  </svg>
                  <span
                    class="mat-focus-indicator"
                  />
                  <span
                    class="mat-mdc-button-touch-target"
                  />
                </button>
                <button
                  aria-label="Next page"
                  class="mdc-icon-button mat-mdc-icon-button mat-mdc-button-base mat-mdc-tooltip-trigger mat-mdc-paginator-navigation-next mat-mdc-button-disabled-interactive mat-unthemed _mat-animation-noopable"
                  disabledinteractive=""
                  mat-ripple-loader-centered=""
                  mat-ripple-loader-class-name="mat-mdc-button-ripple"
                  mat-ripple-loader-uninitialized=""
                  maticonbutton=""
                  mattooltipposition="above"
                  type="button"
                >
                  <span
                    class="mat-mdc-button-persistent-ripple mdc-icon-button__ripple"
                  />
                  <svg
                    aria-hidden="true"
                    class="mat-mdc-paginator-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                    />
                  </svg>
                  <span
                    class="mat-focus-indicator"
                  />
                  <span
                    class="mat-mdc-button-touch-target"
                  />
                </button>
              </div>
            </div>
          </div>
        </mat-paginator>
      </div>
    </section>
  </div>
</app-dashboard>
`;
